fix: bugs by uttamdeep kaur

amitojsingh opened the pull request:
- Audio Auto Play doesn’t start with saved  -> Fix
- Change Audio settings playback speed  -> Fix
- Fix audio player when bottom navigation is on  -> Fix
- fix Audio Track dialog box not opening on other banis. -> Fix
- fix autoPlay issue
- if audio play is selected and when user click on the next from audio track player it won’t play automatically. -> fix

amitojsingh committed:
fix audio functionality

amitojsingh committed:
fix tests

amitojsingh committed:
fix deep scan bugs

amitojsingh committed:
Merge branch 'dev' into fix-functionality

amitojsingh committed:
fix: bugs by uttamdeep kaur

amitojsingh committed:
Merge branch 'fix-functionality' into fix-autoplay

amitojsingh committed:
fix: auto play issue with trackPlayer

amitojsingh committed:
added test suite for AudioTrackDialog

Gauravjeetsingh commented on src/ReaderScreen/components/AudioPlayer/index.jsx:
```diff
@@ -102,14 +101,15 @@ const AudioPlayer = ({ baniID, title, webViewRef }) => {
   useEffect(() => {
-    if (isAudioAutoPlay && currentPlaying) {
+    if (currentPlaying) {
+      console.log("currentPlaying", currentPlaying);
```
We can remove the debug console logs. There are few more in this diff, have a look

amitojsingh replied:
fixed

Gauravjeetsingh commented on src/ReaderScreen/components/AudioPlayer/components/AudioTrackDialog/index.jsx:
Is this correct? The code is using property `displayName` instead of `name`?

amitojsingh replied:
good catch. Thanks I fixed it.

Gauravjeetsingh commented on src/ReaderScreen/components/AudioPlayer/components/AudioControlBar/index.test.jsx:
```diff
@@ -0,0 +1,492 @@
+
+import { render, fireEvent, act, waitFor } from "@testing-library/react-native";
+
+import { useDownloadManager } from "../../hooks"; // <-- adjust path
```
Why is there a comment that says `<-- adjust path` is that a todo for later, or if it's already adjusted, we can remove the comment.

amitojsingh replied:
done

Gauravjeetsingh reviewed (changes requested):
Just some minor changes, have a look

amitojsingh committed:
Merge branch 'dev' into fix-functionality

amitojsingh committed:
Merge branch 'fix-functionality' into fix-autoplay

amitojsingh committed:
Refactor AudioPlayer and AudioTrackDialog components

- Removed console.log statements from AudioPlayer to clean up the code.
- Updated prop types in AudioTrackDialog to include new properties: displayName, audioUrl, lyricsUrl, trackLengthSec, and trackSizeMB for better data structure and validation.
- Adjusted import path in AudioControlBar test file for consistency.

amitojsingh committed:
Merge branch 'dev' into fix-autoplay

Gauravjeetsingh reviewed (approved):
Just some minor changes, have a look
